src/licenses/license.jl

```diff
@@ -0,0 +1,41 @@
+################
+# License Type #
+################
+
+@ghdef mutable struct License
+    key::Union{String, Nothing}
+    name::Union{String, Nothing}
+    spdx_id::Union{String, Nothing}
+    url::Union{HTTP.URI, Nothing}
+    html_url::Union{HTTP.URI, Nothing}
+    description::Union{String, Nothing}
+    permissions::Union{Vector{String}, Nothing}
+    conditions::Union{Vector{String}, Nothing}
+    limitations::Union{Vector{String}, Nothing}
+    body::Union{String, Nothing}
+    featured::Union{Bool, Nothing}
+end
+
+License(spdx_id::AbstractString) = License(Dict("spdx_id" => spdx_id))
+
+namefield(license::License) = license.spdx_id
```

Comment:
I'm not sure what the best choice is here, the obvious options would be `key`, `name`, or `spdx_id`. Some examples:

```json
[
  {
    "key": "mit",
    "name": "MIT License",
    "spdx_id": "MIT"
  },
  {
    "key": "lgpl-3.0",
    "name": "GNU Lesser General Public License v3.0",
    "spdx_id": "LGPL-3.0"
  },
  {
    "key": "mpl-2.0",
    "name": "Mozilla Public License 2.0",
    "spdx_id": "MPL-2.0"
  },
  {
    "key": "agpl-3.0",
    "name": "GNU Affero General Public License v3.0",
    "spdx_id": "AGPL-3.0"
  },
  {
    "key": "unlicense",
    "name": "The Unlicense",
    "spdx_id": "Unlicense"
  },
  {
    "key": "apache-2.0",
    "name": "Apache License 2.0",
    "spdx_id": "Apache-2.0"
  },
  {
    "key": "gpl-3.0",
    "name": "GNU General Public License v3.0",
    "spdx_id": "GPL-3.0"
  }
]
```

Reply:
I think it's only super important when the field is used as an identifier in other API calls. Seems like there aren't any here.

Reply:
A call to `licence` will take it as identifier. Afaict the endpoint takes both `key` and `spdx_id` but not `name`.

Reply:
Ah okay. I think the SPDX ID is fine then!